lib/perl/Genome/InstrumentData/AlignmentResult/Command/CramToBam.pm

```diff
@@ -0,0 +1,47 @@
+package Genome::InstrumentData::AlignmentResult::Command::CramToBam;
+
+use strict;
+use warnings;
+
+use Genome;
+
+class Genome::InstrumentData::AlignmentResult::Command::CramToBam {
+    is => 'Genome::InstrumentData::AlignmentResult::Command::ConvertBase',
+};
+
+sub shortcut {
+    my $self = shift;
+
+    return $self->_is_currently_bam;
+}
+
+sub execute {
+    my $self = shift;
+
+    my $guard = $self->_lock()->unlock_guard;
+
+    return 1 if $self->_is_currently_bam;
+
+    unless($self->_is_currently_cram) {
+        $self->fatal_message("This result cannot be converted to BAM.");
```

Comment:
If it's already BAM we would've hit the "return 1" up above.  This error case protects against a mysterious other format if one happens to exist in the future.